source/fundamentals/authentication.txt

```diff
@@ -142,6 +143,161 @@ string as follow:
+- ``<awsSessionToken>`` - value of the AWS session token
+
+.. tip::
+
+   To learn more about configuring MongoDB Atlas with AWS IAM, see the
+   :atlas:`Set Up Passwordless Authentication with AWS IAM Roles </security/passwordless-authentication/#set-up-passwordless-authentication-with-aws-iam-roles>` guide.
+
+Specify Your AWS IAM Credentials
+++++++++++++++++++++++++++++++++
+
+You can supply your AWS IAM credentials on a ``MongoClientSettings`` object either by 
+using a ``MongoCredential`` object or as part of the connection string. Select the 
+:guilabel:`Connection String` or :guilabel:`MongoCredential` tab to
+see the corresponding syntax for specifying your credentials: 
+
+.. tabs::
+
+   .. tab:: Connection String
+      :tabid: mongodb-aws-mongoclientsettings-connection-string
+      
+      .. code-block:: csharp
+
+         var connectionString = "mongodb+srv://<awsKeyId>:<awsSecretKey>@<atlasUri>?authSource=$external&authMechanism=MONGODB-AWS";
+         var mongoClientSettings = MongoClientSettings.FromConnectionString(connectionString);
+         var client = new MongoClient(mongoClientSettings);
+
+      If you're using an AWS session token, include the ``authMechanismProperties``
+      parameter in the connection string as shown below:
+
+      .. code-block:: csharp
+
+         var connectionString = "mongodb+srv://<awsKeyId>:<awsSecretKey>@<atlasUri>?authSource=$external&authMechanism=MONGODB-AWS&authMechanismProperties=AWS_SESSION_TOKEN:<awsSessionToken>";
+
+   .. tab:: MongoCredential
+      :tabid: mongodb-aws-mongoclientsettings-mongo-credential
+
+      .. code-block:: csharp
+
+         var mongoClientSettings = MongoClientSettings.FromConnectionString("mongodb+srv://<atlasUri>");
+         mongoClientSettings.Credential = new MongoCredential("MONGODB-AWS", new MongoExternalIdentity("<awsKeyId>"), new PasswordEvidence("<awsSecretKey>"));
+         var client = new MongoClient(mongoClientSettings);
+
+      If you're using an AWS session token, call the ``WithMechanismProperty()`` 
+      method on your ``MongoCredential`` object as shown below:
+
+      .. code-block:: csharp
+
+         mongoClientSettings.Credential = new MongoCredential("MONGODB-AWS", new MongoExternalIdentity("<awsKeyId>"), new PasswordEvidence("<awsSecretKey>"))
+             .WithMechanismProperty("AWS_SESSION_TOKEN", "<awsSessionToken>");
+
+Retrieve Credentials Automatically
+++++++++++++++++++++++++++++++++++
+
+Instead of specifying your AWS IAM credentials in ``MongoClientSettings``, you can 
+instruct the {+driver-short+} to use the AWS SDK to automatically retrieve your 
+credentials from an external source. To instruct the driver to
+retrieve your credentials, perform the following actions:
+
+- Specify ``MONGODB-AWS`` as the authentication mechanism
+- Specify that the authentication source is external to MongoDB
+- Set your credentials in the appropriate location 
+
+You can specify the authentication mechanism and database either
```

Comment:
Suggestion:

Same suggestion as [this comment](https://github.com/mongodb/docs-csharp/pull/83#discussion_r1137376720):

```suggestion
You can specify the authentication mechanism and source either
```